packages/snaps-cli/src/cmds/watch/watchHandler.ts

```diff
@@ -32,37 +33,44 @@ export async function watch(argv: YargsArgs): Promise<void> {
     src.indexOf('/') === -1 ? '.' : src.substring(0, src.lastIndexOf('/') + 1);
   const outfilePath = getOutfilePath(dist, outfileName as string);
 
-  const watcher = chokidar.watch(rootDir, {
-    ignoreInitial: true,
-    ignored: [
-      '**/node_modules/**',
-      `**/${dist}/**`,
-      `**/test/**`,
-      `**/tests/**`,
-      `**/*.test.js`,
-      `**/*.test.ts`,
-      /* istanbul ignore next */
-      (str: string) => str !== '.' && str.startsWith('.'),
-    ],
-  });
+  const rebuild = async (path?: string, logMessage?: string) => {
+    if (logMessage !== undefined) {
+      console.log(logMessage);
+    }
 
-  watcher
-    .on('ready', () => {
-      bundle(src, outfilePath, argv);
-    })
-    .on('add', (path: string) => {
-      console.log(`File added: ${path}`);
-      bundle(src, outfilePath, argv);
+    try {
+      await bundle(src, outfilePath, argv);
+      await processManifestCheck(argv);
+      await processEval({ ...argv, bundle: outfilePath });
+    } catch (error) {
+      logError(`Error processing "${path}".`, error);
+    }
```

Comment:
The `watch` command should never fail, but it should reliably output errors to the user.